rainmaker_orchestrator/server.py

```
@@ -1,426 +1,628 @@
+        raise HTTPException(
+            status_code=500,
+            detail=f"Directory creation failed: {e!r}"
+        )
+
+
+# ============================================================================
+# Error Handlers
+# ============================================================================
+
+@app.exception_handler(404)
+async def not_found_handler(request, exc):
```

Ruff (0.14.10)

Unused function argument: `exc`

(ARG001)